### ISSUE TITLE
E0004 does not detected fot tuples

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8384]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4354-213-nightly
* **Rust toolchain version:** 1.57.0 (f1edd0429 2021-11-29) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3.2 (CL-213.6461.75)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description

<img width="675" alt="image" src="https://user-images.githubusercontent.com/49211026/148757592-4ee2e66b-2612-418f-842b-b930baf73f3f.png">

## Steps to reproduce
```rust
fn f(s: Option<i32>, d: Option<bool>) {
    match (s, d) {
        (Some(..), Some(..)) => {}
    }
}
```
